SearchService/solr_interface.py

```diff
@@ -0,0 +1,491 @@
+""" Top level functions for SOLR functions. """
+import calendar
+import logging
+import os
+import simplejson
+import sys
+import urllib2
+
+import query_parser
+import search_exceptions
+
+from datetime import datetime
+
+from query_parser import Document
+
+sys.path.append(os.path.join(os.path.dirname(__file__), "../lib"))
+import appscale_info
+
+sys.path.append(os.path.join(os.path.dirname(__file__), "../AppServer"))
+from google.appengine.datastore.document_pb import FieldValue
+from google.appengine.api.search import search_service_pb
+
+class Solr():
+  """ Class for doing solar operations. """
+
+  # The port SOLR is running on.
+  SOLR_SERVER_PORT = 8983
+
+  def __init__(self):
+    """ Constructor for solr interface. """
+    self._search_location = appscale_info.get_search_location()
+
+  def __get_index_name(self, app_id, namespace, name):
+    """ Gets the internal index name.
+
+    Args:
+      app_id: A str, the application identifier.
+      namespace: A str, the application namespace.
+      name: A str, the index name.
+    Returns:
+      A str, the internal name of the index.
+    """
+    return app_id + "_" + namespace + "_" + name  
+
+  def delete_doc(self, doc_id):
+    """ Deletes a document by doc ID.
+
+    Args:
+      doc_id: A list of document IDs.
+    """
+    solr_request = {"delete": {"id": doc_id}}
+    solr_url = "http://{0}:{1}/solr/update?".format(self._search_location,
```

Comment:
Here's a suggestion:
Make this portion: "http://{0}:{1}/solr/" a constant and append the paths ('/update?...', '/schema/fields, etc.') when needed.